docs/python/index.rst

```diff
@@ -1,7 +1,8 @@
+.. _python_prog_ref:
 
```

Comment:
About the `.jpg` image of ~256KB: I think we will improvise/update such images in the future. Could this binary blob be somewhere else? Not sure where/how but just thinking if it's worth revisiting strategy for adding images to the repo. (as you know, I am cautious bit about growth of repo size esp with binary data 😅 )

Reply:
Oops..I see that there is discussion already above 😄. By pressing `Request change`, I didn't meant to block this PR. But do you think there are any easy alternatives?

Reply:
If it's not in the repo, then the documentation becomes fragmented and fragile (as more things have to work than just the repo to build/view the documentation).

Beyond the scope of this pull request, but: should we consider moving the documentation into a submodule so that it doesn't have to be downloaded every time?

Reply:
> then the documentation becomes fragmented and fragile (as more things have to work than just the repo to build/view the documentation).

Don't have good / definite answer as there are pros/cons with both approaches. We don't want to get fragmented / loose convenience if there is not going to be issue.

My comment was motivated from the thought that whether there could be just simple repo/drive/bucket/or-whathever where we can dump stuff like images and then we can refer those via URLs in the documentation.

> I would argue that the documentation would benefit from having a lot more examples with screenshots. 

Definitely, I agree!


I don't think we need to block this PR. But this is something worth keeping in mind and exploring options.

Reply:
Maybe we can also look into https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github? Screenshots are not very large, but from a quick glance it seems like it would avoid the problem of the overall repo size getting out of hand. I have no first-hand experience of Git-LFS.

Reply:
@alexsavulescu for tutorial zip downloads which besides images are the other main reason we'd have binary blobs: can we host the individual plain text files in the repo and have readthedocs zip it to make it available for bulk download?

Reply:
The downloadable content is generally zips of plain-text files that we may want to edit later.

For one of the optimization tutorials, we created [simtooldb/neuron-optimization-tutorial2](https://github.com/simtooldb/neuron-optimization-tutorial2 ) which has the plain-text files, and the tutorial itself links to that repo's zip download.

The nice thing about that approach is that we never have to store the binary blobs, and we get meaningful diffs.

The inconvenient thing about that approach is that it needs a separate repository for each downloadable entity as you can't (as far as I know) get a link from GitHub for zip downloads of anything other than the entire branch... which is why that went in a separate organization, but that introduces its own problems of file management.

The thing that I just learned I could do better: by default, the commit id ends up in the filename of the zip BUT I just learned you can [specify a download filename in HTML](https://www.w3schools.com/tags/att_a_download.asp) to override the default. (That would work and would be relatively harmless, but it would require adding a `.. raw::` which wouldn't be directly convertible to non-HTML formats. As-is we use that in a couple spots, most notably for the video embeddings.)

Perhaps this discussion should be moved to a separate issue (or GitHub discussion?) since it's mostly independent of this pull request?

Reply:
Yes, we can open another ticket for this

Reply:
I created one at: https://github.com/neuronsimulator/nrn/issues/1767